solver-benchmarks/HackageBenchmark.hs

```diff
@@ -187,7 +185,7 @@ hackageBenchmarkMain = do
           then do
             putStrLn $ "Obtaining the package list (using " ++ argCabal1 ++ ") ..."
             list <- readProcess argCabal1 ["list", "--simple-output"] ""
-            return $ nub [mkPackageName $ head (words line) | line <- lines list]
+            return . nub $ mkPackageName <$> catMaybes [listToMaybe (words line) | line <- lines list]
```

Comment:
This change is unrelated and I happen to disagree with it. I understand the previous version as "a package name from the first word of each line". While I find the use of listToMaybe/catMaybes in the new version less clear.

Is the goal to avoid `head` because it is a partial function? Would something like 
`[ mkPackageName n | n:_ <- map words (lines line) ]` work?

Reply:
The change is related. If I revert it, here's the problem it fixes;

```
$ cabal build solver-benchmarks --enable-tests --enable-benchmarks
...
Resolving dependencies...
Build profile: -w ghc-9.8.1 -O1
In order, the following will be built (use -v for more details):
 - Cabal-syntax-3.11.0.0 (lib) (configuration changed)
 - solver-benchmarks-3 (lib) (configuration changed)
 - solver-benchmarks-3 (test:unit-tests) (configuration changed)
 - solver-benchmarks-3 (exe:hackage-benchmark) (configuration changed)
Configuring library for Cabal-syntax-3.11.0.0...
Preprocessing library for Cabal-syntax-3.11.0.0...
Building library for Cabal-syntax-3.11.0.0...
Configuring library for solver-benchmarks-3...
Preprocessing library for solver-benchmarks-3...
Building library for solver-benchmarks-3...
[1 of 1] Compiling HackageBenchmark

HackageBenchmark.hs:188:43: error: [GHC-63394] [-Wx-partial, -Werror=x-partial]
    In the use of ‘head’
    (imported from Prelude, but defined in GHC.List):
    "This is a partial function, it throws an error on empty lists.
    Use pattern matching or Data.List.uncons instead.
    Consider refactoring to use Data.List.NonEmpty."
    |
188 |             return $ nub [mkPackageName $ head (words line) | line <- lines list]
    |                                           ^^^^

HackageBenchmark.hs:340:44: error: [GHC-63394] [-Wx-partial, -Werror=x-partial]
    In the use of ‘head’
    (imported from Prelude, but defined in GHC.List):
    "This is a partial function, it throws an error on empty lists.
    Use pattern matching or Data.List.uncons instead.
    Consider refactoring to use Data.List.NonEmpty."
    |
340 |   | allEqual rs                          = head rs
    |                                            ^^^^
Error: [Cabal-7125]
Failed to build solver-benchmarks-3
(which is required by test:unit-tests from solver-benchmarks-3 and exe:hackage-benchmark
from solver-benchmarks-3).
```

Reply:
Thanks for your suggestion that I like a lot.

I'm a fan of list comprehensions but haven't been using partial matches with them. Wow, have I been missing out!

Reply:
I've gone ahead an used a partial pattern for filtering.